src/test.py

```diff
@@ -62,10 +65,105 @@ def wrap_socket(self, s, server_hostname):
     def patch(cls):
         return mock.patch("ssl.create_default_context", wraps=cls)
 
+class SilentTk:
+    def bind(self, event, callback):
+        pass
+
+tkinter.Tk = SilentTk
+
+class SilentCanvas:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_text(self, x, y, text, font=None, anchor=None):
+        pass
+
+    def pack(self):
+        pass
+
+    def delete(self, v):
+        pass
+
+tkinter.Canvas = SilentCanvas
+
+class MockCanvas:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_text(self, x, y, text, font=None, anchor=None):
+        if font or anchor:
+            print("create_text: x={} y={} text={} font={} anchor={}".format(
+                x, y, text, font, anchor))
+        else:
+            print("create_text: x={} y={} text={}".format(
+                x, y, text))
+
+    def pack(self):
+        pass
+
+    def delete(self, v):
+        pass
+
+original_tkinter_canvas = tkinter.Canvas
+
+def patch_canvas():
+    tkinter.Canvas = MockCanvas
+
+def unpatch_canvas():
+    tkinter.Canvas = original_tkinter_canvas
+
+class MockFont:
+    def __init__(self, size=None, weight=None, slant=None, style=None):
+        self.size = size
+        self.weight = weight
+        self.slant = slant
+        self.style = style
+
+    def measure(self, word):
+        return self.size * len(word)
+
+    def metrics(self, name=None):
+        all = {"ascent" : self.size / 2, "descent": self.size / 2,
+            "linespace": self.size * 2}
+        if name:
+            return all[name]
+        return all
+
+    def __repr__(self):
+            return "Font size={} weight={} slant={} style={}".format(
+                self.size, self.weight, self.slant, self.style)
+
+tkinter.font.Font = MockFont
+
 def errors(f, *args, **kwargs):
     try:
         f(*args, **kwargs)
     except Exception as e:
         return True
     else:
         return False
+
+def breakpoint(name, value1=None, value2=None, value3=None):
+    if value3:
+        print('breakpoint: name={} value1={} value2={} value3={}'.format(
+            name, value1, value2, value3))
+        return
+
+    if value2:
+        print('breakpoint: name={} value1={} value2={}'.format(
+            name, value1, value2))
+        return
+
+    if value1:
+        print('breakpoint: name={} value1={}'.format(name, value1))
+        return
+
+    print('breakpoint: name={}'.format(name))
```

Comment:
```suggestion
def breakpoint(name, **kwargs):
    kwstr = (", " + ", ".join(["{}={}".format(k, v) for k, v in kwargs.items()]) if kwargs else ""
    print('breakpoint: name={}{}'.format(name, kwstr))
```

There's a Python trick here for arbitrary keyword arguments. I have a comment like the one Text/Tag about making the output easy to group by keeping quotes around some of the strings and so on, but it's not that important.